src/main/kotlin/org/rust/ide/template/macros/utils.kt

```diff
@@ -0,0 +1,15 @@
+package org.rust.ide.template.macros
+
+import com.intellij.psi.PsiElement
+import org.rust.lang.core.psi.RustCompositeElement
+import org.rust.lang.core.psi.RustPatBindingElement
+import org.rust.lang.core.resolve.RustResolveEngine
+
+fun getPatBindingsVisibleAt(place: PsiElement?): List<RustPatBindingElement> {
+    if (place == null) return emptyList()
+    val pivot = place as? RustCompositeElement
+    return RustResolveEngine.enumerateScopesFor(place)
+        .flatMap { RustResolveEngine.declarations(it, pivot) }
+        .mapNotNull { it as? RustPatBindingElement }
+        .toList()
```

Comment:
> Can this limitation be removed?

Yes. 

> This is incredibly slow

Well, you are asking for all visible symbols (even for the private ones) :) I think that you can add a `takeWhile` after `enumerateScopesFor` to consider only local scopes.